### PR TITLE
cli: move helm and terraform out of kubernetes package

### PR DESCRIPTION
**Pull request by daniel-weisse**
### Context
The `kubernetes` package of the CLI currently has dependencies on packages it shouldn't have, as well as offering a public interface way larger in scope than necessary.
More specifically, the `cmd` package of the CLI for most of its use cases calls into the `kubernetes` to perform actions not directly associated with Kubernetes itself. This mostly affects the `terraform`, and `helm` package, both of which are called through the `kubernetes` package in many cases.

### Proposed change(s)

- Remove `terraform` and `helm` from the `kubernetes` package

### Additional info
There will be a follow up PR to further clean up the package. Mostly in regards to the usage of `StableClient` and `DynamicClient`, as well as renaming a couple things.

### CODE REVIEW

File: cli/internal/cmd/upgrade.go [hunk 1]
```diff
@@ -23,3 +27,24 @@ func NewUpgradeCmd() *cobra.Command {
...
+type upgradeCmdKind int
+
+const (
+	// upgradeCmdKindCheck corresponds to the upgrade check command.
+	upgradeCmdKindCheck upgradeCmdKind = iota
+	// upgradeCmdKindApply corresponds to the upgrade apply command.
+	upgradeCmdKindApply
+)
+
+func generateUpgradeID(kind upgradeCmdKind) string {
```
Comment (elchead) [inline cli/internal/cmd/upgrade.go:41]: nit: should iam upgrade use this too?

Comment (daniel-weisse) [reply cli/internal/cmd/upgrade.go:41]: I applied this suggestion, but this also means IAM upgrades are placed in folders similar named to other upgrades.
Is this something we are OK with, or should be adjust this to stay as `iam-<uid>` ?

Comment (daniel-weisse) [reply cli/internal/cmd/upgrade.go:43]: Added a new `upgradeCmdKind` for IAM

**Review by elchead — APPROVED** [review]

**Review by derpsteb — APPROVED** [review]
I would wait with merging until v2.10 is out.

**Pull request merged**